File: src/main/java/com/back/together02be/chart/dto/response/ChartRes.java
package com.back.together02be.chart.dto.response;

import java.util.List;

import com.back.together02be.chart.constant.ChartPeriod;
import com.back.together02be.chart.dto.Candle;

public record ChartRes(
	String stockCode,
	String name,
	String period,
	String interval,
	List<CandleRes> candles
) {
	public static ChartRes of(String stockCode, String name, ChartPeriod period, List<Candle> candles) {
		List<CandleRes> dtos = candles.stream().map(CandleRes::from).toList();
		return new ChartRes(stockCode, name, period.getValue(),
			period == ChartPeriod.THREE_MONTHS ? "DAY" : "WEEK", dtos);
	}
}

File: src/main/kotlin/com/back/together02be/chart/dto/response/ChartRes.kt
package com.back.together02be.chart.dto.response

import com.back.together02be.chart.constant.ChartPeriod
import com.back.together02be.chart.dto.Candle


data class ChartRes(
    val stockCode: String,
    val name: String,
    val period: String,
    val interval: String,
    val candles: List<CandleRes>
) {

    companion object {

        fun of(
            stockCode: String,
            name: String,
            period: ChartPeriod,
            candles: List<Candle>,
        ): ChartRes {
            val dtos = candles.map(CandleRes::from)
            return ChartRes(
                stockCode = stockCode,
                name = name,
                period = period.value,
                interval = if (period == ChartPeriod.THREE_MONTHS) "DAY" else "WEEK",
                candles = dtos,
            )
        }
    }
}
